BBallStatsTrack/src/com/example/bballstatstrack/models/utils/JSONDeserializer.java

```diff
@@ -0,0 +1,431 @@
+        int made3pt = Integer.MIN_VALUE;
+        int offReb = Integer.MIN_VALUE;
+        int defReb = Integer.MIN_VALUE;
+        int assist = Integer.MIN_VALUE;
+        int to = Integer.MIN_VALUE;
+        int stl = Integer.MIN_VALUE;
+        int blk = Integer.MIN_VALUE;
+        int foul = Integer.MIN_VALUE;
+        int playingTimeSec = Integer.MIN_VALUE;
+        reader.beginObject();
+        String name;
+        while( reader.hasNext() )
+        {
+            name = reader.nextName();
+            if( name.equals( PlayerStats.NUMBER.toString() ) )
+            {
+                number = reader.nextInt();
+            }
+            else if( name.equals( PlayerStats.NAME.toString() ) )
+            {
+                fullName = reader.nextString();
+            }
+            else if( name.equals( PlayerStats.MISS_1PT.toString() ) )
+            {
+                miss1pt = reader.nextInt();
+            }
+            else if( name.equals( PlayerStats.MISS_2PT.toString() ) )
+            {
+                miss2pt = reader.nextInt();
+            }
+            else if( name.equals( PlayerStats.MISS_3PT.toString() ) )
+            {
+                miss3pt = reader.nextInt();
+            }
+            else if( name.equals( PlayerStats.MADE_1PT.toString() ) )
+            {
+                made1pt = reader.nextInt();
+            }
+            else if( name.equals( PlayerStats.MADE_2PT.toString() ) )
+            {
+                made2pt = reader.nextInt();
+            }
+            else if( name.equals( PlayerStats.MADE_3PT.toString() ) )
+            {
+                made3pt = reader.nextInt();
+            }
+            else if( name.equals( PlayerStats.OFFENSIVE_REBOUND.toString() ) )
+            {
+                offReb = reader.nextInt();
+            }
+            else if( name.equals( PlayerStats.DEFENSIVE_REBOUND.toString() ) )
+            {
+                defReb = reader.nextInt();
+            }
+            else if( name.equals( PlayerStats.ASSIST.toString() ) )
+            {
+                assist = reader.nextInt();
+            }
+            else if( name.equals( PlayerStats.TURNOVER.toString() ) )
+            {
+                to = reader.nextInt();
+            }
+            else if( name.equals( PlayerStats.STEAL.toString() ) )
+            {
+                stl = reader.nextInt();
+            }
+            else if( name.equals( PlayerStats.BLOCK.toString() ) )
+            {
+                blk = reader.nextInt();
+            }
+            else if( name.equals( PlayerStats.FOUL.toString() ) )
+            {
+                foul = reader.nextInt();
+            }
+            else if( name.equals( PlayerStats.PLAYING_TIME.toString() ) )
+            {
+                playingTimeSec = reader.nextInt();
+            }
+        }
+        reader.endObject();
+        return new Player( number, fullName, miss1pt, miss2pt, miss3pt, made1pt, made2pt, made3pt, offReb, defReb,
```

Comment:
If you can implement Builder Pattern you can change this to Object.BuilderMethod().build() which will pass the needed object.